addons/udes_stock/tests/test_stock_move_line.py

```diff
@@ -540,6 +540,795 @@ def test_over_mark_error(self):
+
+        product_ids = [{"barcode": "productTangerine", "uom_qty": 10, "lot_names": ["2"]}]
+        location = self.test_stock_location_01
+        pallet_to_pick_onto = self.create_package(name="UDES1")
+
+        res = first_picking.move_line_ids.prepare(
+            product_ids=product_ids, location=location, result_package=pallet_to_pick_onto
+        )
+        for mls, mls_values in res.items():
+            mls.mark_as_done(mls_values)
+        self.assertEqual(first_picking.move_line_ids.lot_id.name, "2")
+        self.assertEqual(first_picking.move_line_ids.product_uom_qty, 10.0)
+        self.assertEqual(first_picking.move_line_ids.qty_done, 10.0)
+        self.assertEqual(second_picking.move_line_ids.lot_id.name, "1")
+        self.assertEqual(second_picking.move_line_ids.product_uom_qty, 10.0)
+        self.assertEqual(second_picking.move_line_ids.qty_done, 0.0)
+
+        self.assertEqual(Quant.search([("lot_id.name", "=", "1")]).reserved_quantity, 10.0)
+        self.assertEqual(Quant.search([("lot_id.name", "=", "2")]).reserved_quantity, 10.0)
+    
+    def test_swapping_the_lots_on_two_move_lines_with_partial_qty(self):
+        """
+        Swap partial quantity of lot when new lot is actually reserved on the move_line of another picking
+        """
+        Quant = self.env["stock.quant"]
+
+        self.picking_type_pick.u_allow_swapping_tracked_products = True
+        for lot_num in range(1, 6):
+            self.create_quant(
+                self.tangerine.id,
+                location_id=self.test_stock_location_01.id,
+                qty=10,
+                lot_name=f"{lot_num}",
+            )
+        product_info = [{"product": self.tangerine, "uom_qty": 10}]
+        first_picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+        second_picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+        self.assertEqual(first_picking.move_line_ids.lot_id.name, "1")
+        self.assertEqual(second_picking.move_line_ids.lot_id.name, "2")
+
+        product_ids = [{"barcode": "productTangerine", "uom_qty": 5, "lot_names": ["2"]}]
+        location = self.test_stock_location_01
+        pallet_to_pick_onto = self.create_package(name="UDES1")
+
+        res = first_picking.move_line_ids.prepare(
+            product_ids=product_ids, location=location, result_package=pallet_to_pick_onto
+        )
+        for mls, mls_values in res.items():
+            mls.mark_as_done(mls_values)
+        self.assertEqual(second_picking.move_line_ids.mapped("lot_id.name"), ["2","1"])
+        self.assertEqual(second_picking.move_line_ids.mapped("product_uom_qty"), [5.0, 5.0])
+        self.assertEqual(second_picking.move_line_ids.mapped("qty_done"), [0.0,0.0])
+
+        self.assertEqual(Quant.search([("lot_id.name", "=", "1")]).reserved_quantity, 10.0)
```

Comment:
We could use a helper function that checks for a given lot name what the reserved quantity should be. Then use it in all the tests, since the previous ones we are only checking if the move line info has been swapped. Also consider checking the reserved quantity of the original lot before the swap.